File: app/src/main/java/com/kickstart/application/ui/BaseActivity.java
package com.kickstart.application.ui;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import dagger.android.AndroidInjection;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasDispatchingSupportFragmentInjector;

public abstract class BaseActivity extends AppCompatActivity implements HasDispatchingSupportFragmentInjector {
    @Inject
    DispatchingAndroidInjector<Fragment> mFragmentInjector;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState, @Nullable PersistableBundle persistentState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState, persistentState);
    }

    @Override
    public DispatchingAndroidInjector<Fragment> supportFragmentInjector() {
        return mFragmentInjector;
    }
}

File: app/src/main/java/com/kickstart/railway/ui/BaseActivity.kt
package com.kickstart.railway.ui

import android.arch.lifecycle.LifecycleActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.support.v4.app.Fragment

import javax.inject.Inject

import dagger.android.AndroidInjection
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasDispatchingSupportFragmentInjector

abstract class BaseActivity : LifecycleActivity(), HasDispatchingSupportFragmentInjector {
    @Inject
    lateinit internal var mFragmentInjector: DispatchingAndroidInjector<Fragment>

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState, persistentState)
    }

    override fun supportFragmentInjector(): DispatchingAndroidInjector<Fragment> {
        return mFragmentInjector
    }
}
